src/controllers/connectionManager.ts

```diff
@@ -314,6 +312,13 @@ export default class ConnectionManager {
             let mruConnection: IConnectionInfo = <any>{};
 
             if (Utils.isNotEmpty(result.connectionId)) {
+                // Use the original connection information to save the MRU connection.
+                // for connections that a database is not provided, the database information will be updated
+                // to the default database name, if we use the new information as the MRU connection,
+                // the connection information will be different from the saved connections (saved connection's database property is empty).
+                // When deleting the saved connection, we won't be able to find its corresponding recent connection,
+                // and the saved connection credentials will become orphaned.
+                mruConnection = Object.assign({}, connection.credentials);
```

Comment:
I'd consider doing a deep clone here just to be safe...right now the credentials only holds base types I think but if we add more complex properties later on it would be good not to have to remember to update this as well.